Code/GraphMol/SmilesParse/SmilesParseOps.cpp

```diff
@@ -25,8 +25,12 @@ void CheckRingClosureBranchStatus(RDKit::Atom *atom, RDKit::RWMol *mp) {
   // the stereochem is wrong.
   // detect the branch (= atom isn't the last one)
   // and reverse the stereochem if the atom has chiral stereochemistry
-  // and is currently degree two (protects against C1CN[C@](O)(N)1)
-  if (atom->getIdx() != mp->getNumAtoms(true) - 1 && atom->getDegree() == 2 &&
+  // and is currently degree one   [C@@](F)1(C)CCO1
+  //     or currently degree two   C1CN[C@](O)(N)1
+  //        and not the first atom [C@](Cl)(F)1CC[C@H](F)CC1
+  if (atom->getIdx() != mp->getNumAtoms(true) - 1 &&
+      (atom->getDegree() == 1 ||
+       (atom->getDegree() == 2 && atom->getIdx() != 0)) &&
```

Comment:
Isn't the first example is degree three (no hydrogens) no?

This one: [C@@](F)1(C)CCO1

Reply:
The way to interpret the comment is, 'the degree when the closure digit is encountered during parsing'.

```[C@@](F)1(C)CCO1```

So when the `1` digit is encountered in this instance there is only one neighbor to the carbon atom `[C@@](F)`.

CheckRingClosureBranchStatus is called directly by the yacc parser in blocks like this: 

```
| mol ring_number {
  RWMol * mp = (*molList)[$$];
  Atom *atom=mp->getActiveAtom();
  mp->setAtomBookmark(atom,$2);

  Bond *newB = mp->createPartialBond(atom->getIdx(),
                     Bond::UNSPECIFIED);
  mp->setBondBookmark(newB,$2);
  newB->setProp(RDKit::common_properties::_unspecifiedOrder,1);

  SmilesParseOps::CheckRingClosureBranchStatus(atom,mp);

  INT_VECT tmp;
  atom->getPropIfPresent(RDKit::common_properties::_RingClosures,tmp);
  tmp.push_back(-($2+1));
  atom->setProp(RDKit::common_properties::_RingClosures,tmp);

}
```

The other tokens that call it are: 

```
| mol BOND_TOKEN ring_number {
...
| mol MINUS_TOKEN ring_number {
```

Reply:
@greglandrum Could we make the comment more explicit in the code?  I understand the intent better now, but @coleb had to explain it to me.

Reply:
Makes sense. What about:
```
  // github #786 and #1652: if the ring closure comes after a branch,
  // the stereochem is wrong.
  // This function is called while after closing a branch during construction of
  // the molecule from SMILES and corrects for what happens when parsing odd
  // (and arguably wrong) SMILES constructs like :
  //   1) [C@@](F)1(C)CCO1
  //   2) C1CN[C@](O)(N)1
  //   3) [C@](Cl)(F)1CC[C@H](F)CC1
  // In each of the first two cases the stereochemistry at the chiral atom
  // needs to be reversed. In the third case the stereochemistry should be
  // reversed when the Cl is added, but left alone with the F is added.
  // We recognize these situations using the degree and index of the chiral atom
```

Reply:
I've updated the comment to the above while fixing up the english a little and expounding a tad bit at the end. Let me know what you think.

Reply:
@greglandrum @coleb looks good to me.

Reply:
Merge please, actually waiting on this one to push some other work forward. :-)